### PR TITLE
Allow files to be wrapped with a mutator during add

### PR DESCRIPTION
**Pull request by whyrusleeping**
I'm playing around with different ways of importing files in a side project, and this small change allows me to encrypt or compress files on the fly. 

License: MIT
Signed-off-by: Whyrusleeping <why@ipfs.com>

### CODE REVIEW

File: core/coreunix/add.go [hunk 1]
```diff
@@ -71,26 +71,27 @@ func NewAdder(ctx context.Context, p pin.Pinner, bs bstore.GCBlockstore, ds ipld
...
+	Progress    bool
+	Hidden      bool
+	Pin         bool
+	Trickle     bool
+	RawLeaves   bool
+	Silent      bool
+	Wrap        bool
+	NoCopy      bool
+	Chunker     string
+	FileWrapper func(io.Reader) (io.Reader, error)
```
Comment (schomatis) [inline core/coreunix/add.go]: Why are all the field listed as modified? The `FileWrapper` is the only new field right? Also, just to be extra annoying, could we add a comment explaining the function the newly added field? I'm a strong believer in the broken windows theory :)

File: core/coreunix/add.go [hunk 2]
```diff
@@ -114,6 +115,14 @@ func (adder *Adder) SetMfsRoot(r *mfs.Root) {
 
 // Constructs a node from reader's data, and adds it. Doesn't pin.
 func (adder *Adder) add(reader io.Reader) (ipld.Node, error) {
+	if adder.FileWrapper != nil {
+		wr, err := adder.FileWrapper(reader)
```
Comment (schomatis) [inline core/coreunix/add.go]: So, this is the single point of entry of a reader to the chunker right? (The other similar calls I'm finding in the code are for testing purposes only.)

File: core/coreunix/add.go (hunk 1 above)
Comment (Stebalien) [reply core/coreunix/add.go]: That's gofmt. It aligns everything.

**Comment by magik6k** [comment]
After the files refactor (https://github.com/ipfs/go-ipfs/pull/5661) lands, it should be fairly easy to just wrap file interfaces to do this without having to modify the adder (and it will work via coreapi without touching internals)

**Comment by Stebalien** [comment]
Closing in favor of a `go-ipfs-files` adapter. We can reopen this if that doesn't work.

**Pull request closed without merging**